File: source/reference-ecommerce/stack/shopify-powered/test/shopify-powered-stack.test.ts
import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as ShopifyPoweredStack from '../lib/shopify-powered-stack';

test('API Gateway Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShopifyPoweredStack.ShopifyPoweredStack(app, 'MyTestStack');
  // THEN
  const template = Template.fromStack(stack);

  template.hasResourceProperties('AWS::ApiGateway::RestApi', {
    Name: 'Shopify Powered API'
  });
});

test('Lambda Functions Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShopifyPoweredStack.ShopifyPoweredStack(app, 'MyTestStack');
  // THEN
  const template = Template.fromStack(stack);

  template.resourceCountIs('AWS::Lambda::Function', 3);
});

test('API Gateway Resources Created', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShopifyPoweredStack.ShopifyPoweredStack(app, 'MyTestStack');
  // THEN
  const template = Template.fromStack(stack);

  template.resourceCountIs('AWS::ApiGateway::Resource', 6);  // 3 main resources + 3 proxy resources
});

test('Lambda Permissions Granted', () => {
  const app = new cdk.App();
  // WHEN
  const stack = new ShopifyPoweredStack.ShopifyPoweredStack(app, 'MyTestStack');
  // THEN
  const template = Template.fromStack(stack);

  template.resourceCountIs('AWS::Lambda::Permission', 3);
});

File: converted-code/ShopifyPoweredStackTest.java
import org.junit.jupiter.api.Test;
import software.amazon.awscdk.App;
import software.amazon.awscdk.assertions.Template;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ShopifyPoweredStackTest {

    @Test
    void testApiGatewayCreated() {
        App app = new App();
        ShopifyPoweredStack stack = new ShopifyPoweredStack(app, "MyTestStack");
        Template template = Template.fromStack(stack);

        template.hasResourceProperties("AWS::ApiGateway::RestApi", Map.of(
                "Name", "Shopify Powered API"
        ));
    }

    @Test
    void testLambdaFunctionsCreated() {
        App app = new App();
        ShopifyPoweredStack stack = new ShopifyPoweredStack(app, "MyTestStack");
        Template template = Template.fromStack(stack);

        assertEquals(3, template.resourceCountIs("AWS::Lambda::Function"));
    }

    @Test
    void testApiGatewayResourcesCreated() {
        App app = new App();
        ShopifyPoweredStack stack = new ShopifyPoweredStack(app, "MyTestStack");
        Template template = Template.fromStack(stack);

        assertEquals(6, template.resourceCountIs("AWS::ApiGateway::Resource")); // 3 main resources + 3 proxy resources
    }

    @Test
    void testLambdaPermissionsGranted() {
        App app = new App();
        ShopifyPoweredStack stack = new ShopifyPoweredStack(app, "MyTestStack");
        Template template = Template.fromStack(stack);

        assertEquals(3, template.resourceCountIs("AWS::Lambda::Permission"));
    }
}